smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/middlewares/OperationInputHeadersMiddleware.kt

```diff
@@ -0,0 +1,30 @@
+package software.amazon.smithy.swift.codegen.integration.middlewares
+
+import software.amazon.smithy.model.shapes.OperationShape
+import software.amazon.smithy.model.shapes.ServiceShape
+import software.amazon.smithy.swift.codegen.SwiftWriter
+import software.amazon.smithy.swift.codegen.integration.ProtocolGenerator
+import software.amazon.smithy.swift.codegen.middleware.MiddlewarePosition
+import software.amazon.smithy.swift.codegen.middleware.MiddlewareRenderable
+import software.amazon.smithy.swift.codegen.middleware.MiddlewareStep
+
+class OperationInputHeadersMiddleware : MiddlewareRenderable {
+
+    override val name = "OperationInputHeadersMiddleware"
+
+    override val middlewareStep = MiddlewareStep.SERIALIZESTEP
+
+    override val position = MiddlewarePosition.AFTER
+
+    override fun render(
+        ctx: ProtocolGenerator.GenerationContext,
+        writer: SwiftWriter,
+        serviceShape: ServiceShape,
+        op: OperationShape,
+        operationStackName: String
+    ) {
+        val inputShape = ctx.model.expectShape(op.input.get())
+        val inputShapeName = ctx.symbolProvider.toSymbol(inputShape).name
```

Comment:
update to use ServiceGeneration function please

Reply:
done.